apps/root/src/state/balances/actions.ts

```diff
@@ -0,0 +1,133 @@
+import { createAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { CurrentPriceForChainResponse, Token, TokenList } from '@types';
+import { BalancesState, TokenBalancesAndPrices } from './reducer';
+import { ExtraArgument, RootState } from '@state';
+import { BigNumber } from 'ethers';
+import { NETWORKS } from '@constants';
+import { flatten, keyBy, set } from 'lodash';
+
+export const setLoadingChainPrices = createAction<{ chainId: number; isLoading: boolean }>(
+  'balances/setLoadingChainPrices'
+);
+
+export const fetchWalletBalancesForChain = createAsyncThunk<
+  { chainId: number; tokenBalances: TokenBalancesAndPrices },
+  { tokenList: TokenList; chainId: number; walletAddress: string },
+  { extra: ExtraArgument }
+>('balances/fetchBalancesForChain', async ({ tokenList, chainId, walletAddress }, { extra: { web3Service } }) => {
+  const sdkService = web3Service.getSdkService();
+  const tokens = Object.values(tokenList);
+
+  let balances: Record<number, Record<string, BigNumber>> = {};
+  try {
+    balances = await sdkService.getMultipleBalances(tokens, walletAddress);
+  } catch (e) {
+    console.error(e);
+  }
+
+  const tokenAccountBalances: TokenBalancesAndPrices = {};
+  Object.entries(balances[chainId]).forEach(([tokenAddress, balance]) => {
+    if (balance.gt(0)) {
+      tokenAccountBalances[tokenAddress] = {
+        token: tokenList[tokenAddress],
+        balances: {
+          [walletAddress]: balance,
+        },
+      };
+    }
+  });
+
+  return { chainId, tokenBalances: tokenAccountBalances };
+});
+
+export const fetchPricesForChain = createAsyncThunk<
+  { chainId: number; prices: CurrentPriceForChainResponse },
+  { chainId: number },
+  { extra: ExtraArgument }
+>('prices/fetchPricesForChain', async ({ chainId }, { extra: { web3Service }, getState, dispatch }) => {
+  const sdkService = web3Service.getSdkService();
+  const state = getState() as RootState;
```

Comment:
This is a tricky situation because we've defined the `RootState` type based on the return value of `getState`. In this case we're trying to define the `getState` type using `RootState`, leading to a circular dependency issue. Despite spending some time trying to resolve this, I haven't found a solution yet. Perhaps we should consider moving on and address it in the future?

Reply:
Sure, let me try it tho